examples/units/basic_units.py

```diff
@@ -345,7 +348,20 @@ def convert(val, unit, axis):
         if units.ConversionInterface.is_numlike(val):
             return val
         if np.iterable(val):
-            return [thisval.convert_to(unit).get_value() for thisval in val]
+            if np.ma.isMaskedArray(val):
+                val = val.astype(float).filled(np.nan)
+            out = np.empty((len(val),),  dtype=float)
+            for i, thisval in enumerate(val):
+                if np.ma.is_masked(thisval):
+                    out[i] = np.nan
+                else:
+                    try:
+                        out[i] = thisval.convert_to(unit).get_value()
+                    except AttributeError:
+                        out[i] = thisval
+            return out
+        if np.ma.is_masked(val):
```

Comment:
I think `val` is scalar at that point and then it's equivalent to `val == np.ma.masked`? (not sure)

Reply:
It's not equivalent:
```
In [15]: x = np.ma.array([0, 1], mask=[False, True])

In [16]: np.ma.is_masked(x[0])
False

In [17]: x[0] == np.ma.masked
masked

In [18]: np.ma.is_masked(x[1])
True

In [19]: x[1] == np.ma.masked
masked

```

Reply:
But there is a check for np.iterable() in the block just above with early return, so here val is not an interable but a scalar.

Reply:
That's the point.  In my example above I am using indexing to extract the masked scalar, which canot usefully be compared to np.ma.masked, but must instead be tested using is_masked.

Reply:
Oh, I see, it's that one just cannot compare with np.ma.masked using equality (because it behaves a bit like nan); one has to use is_masked similarly to isnan.
Thanks for clarifying.